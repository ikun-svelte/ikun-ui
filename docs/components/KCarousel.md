---
title: KCarousel
lang: en-US
---

# KCarousel

A carousel component

## Install

::: code-group

```bash [pnpm]
pnpm add @ikun-ui/carousel
```

```bash [yarn]
yarn add @ikun-ui/carousel
```

```bash [npm]
npm install @ikun-ui/carousel
```

:::

## Basic usage

Basic usage.

<demo src="carousel/basic.svelte"  github='Carousel'></demo>

## Indicator

Indicators can be displayed inside the carousel

<demo src="carousel/indicator.svelte" github='Carousel'></demo>

## Arrows

The arrow property determines when arrows are displayed.

<demo src="carousel/arrows.svelte" github='Carousel'></demo>

## Auto height

When the height property is not set, the carousel component dynamically changes height according to the content.

<demo src="carousel/height.svelte" github='Carousel'></demo>

## Custom indicator and arrow

Sometimes you may want to customize indicator and arrow.

<demo src="carousel/custom.svelte" github='Carousel'></demo>

## Carousel Props

| Name         | Type                             | Default | Description                                                                                  |
| ------------ | -------------------------------- | ------- | -------------------------------------------------------------------------------------------- |
| height       | `number`                         | `'-`    | Height of the carousel.                                                                      |
| initialIndex | `number`                         | `0`     | The default display index of carousel (starting from 0).                                     |
| count        | `number`                         | `0`     | Specifies the number of carousels to display, it is required.                                |
| arrow        | `'always' \| 'hover' \| 'never'` | `hover` | How the arrow paginator is displayed.                                                        |
| loop         | `boolean`                        | `true`  | Whether to loop play.                                                                        |
| autoplay     | `boolean`                        | `true`  | Whether to play through automatic page turning.                                              |
| pauseOnHover | `boolean`                        | `true`  | When the mouse moves over the carousel, the automatic page turning playback will be paused.. |
| interval     | `number`                         | `3000`  | Automatic page turning playback interval (unit ms).                                          |
| cls          | `string`                         | `''`    | Additional class for component                                                               |
| attrs        | `Record<string, string>`         | `{}`    | Additional attributes                                                                        |

## Carousel Events

| Name   | Description                        | Type                                                |
| ------ | ---------------------------------- | --------------------------------------------------- |
| change | Triggered when a page turn occurs. | `(data: { index: number, oldIndex:number })=> void` |

## Carousel Slots

| Name       | Description                                                                       |
| ---------- | --------------------------------------------------------------------------------- |
| default    | Customize default content                                                         |
| indicators | Customize indicators content, Use with KIndicators to implement custom indicators |
| arrow      | Customize arrow content, Use with KCarouselArrow to implement a customized arrow  |
