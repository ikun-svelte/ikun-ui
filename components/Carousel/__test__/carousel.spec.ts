import { afterEach, expect, test, describe, beforeEach, vi } from 'vitest';
import KCarousel from '../src';
import KCarouselInit from './fixture/init.svelte';
import KCarouselTriggerClick from './fixture/trigger.click.svelte';
import KCarouselTriggerHover from './fixture/trigger.hover.svelte';
import KCarouselAlways from './fixture/arrow.always.svelte';
import KCarouselHover from './fixture/arrow.hover.svelte';
import KCarouselNever from './fixture/arrow.never.svelte';
import KCarouselLoop from './fixture/loop.svelte';
import KCarouselPlay from './fixture/play.svelte';
import KCarouselApiNext from './fixture/api.next.svelte';
import KCarouselApiPrev from './fixture/api.prev.svelte';
import KCarouselApiGoto from './fixture/api.goto.svelte';
import KCarouselEvtApi from './fixture/event.api.svelte';
import KCarouselEvt from './fixture/event.svelte';
import KCarouselEvtPlay from './fixture/event.play.svelte';
import KCarouselCustom from './fixture/custom.svelte';
import { tick } from 'svelte';
let host;

const initHost = () => {
	host = globalThis.document.createElement('div');
	host.setAttribute('id', 'host');
	globalThis.document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
	vi.useFakeTimers();
});
afterEach(() => {
	host.remove();
	vi.restoreAllMocks();
});

describe('Test: KCarousel', () => {
	vi.mock('svelte', async () => {
		const actual = (await vi.importActual('svelte')) as object;
		return {
			...actual,
			// @ts-ignore
			onMount: (await import('svelte/internal')).onMount
		};
	});
	test('props: cls', async () => {
		// @ts-ignore
		const instance = new KCarousel({
			target: host,
			props: {
				count: 3,
				initialIndex: 0,
				cls: 'k-carousel--test'
			}
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('k-carousel--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: initialIndex', async () => {
		// @ts-ignore
		const instance = new KCarouselInit({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('[data-active="1"]')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: trigger indicators click', async () => {
		// @ts-ignore
		const instance = new KCarouselTriggerClick({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('[data-active="1"]')).toBeTruthy();
		const indicatorsElms = host.querySelectorAll('.k-indicators-item');
		indicatorsElms[2].click();
		await tick();
		expect(host.querySelector('[data-active="2"]')).toBeTruthy();
	});

	test('props: trigger indicators hover', async () => {
		// @ts-ignore
		const instance = new KCarouselTriggerHover({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('[data-active="1"]')).toBeTruthy();
		const indicatorsElms = host.querySelectorAll('.k-indicators-item');
		indicatorsElms[2].dispatchEvent(new Event('mouseenter', { bubbles: true }));
		await tick();
		expect(host.querySelector('[data-active="2"]')).toBeTruthy();
	});

	test('props: arrow is always', async () => {
		// @ts-ignore
		const instance = new KCarouselAlways({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('.k-carousel-arrow-prev')).toBeTruthy();
		expect(host.querySelector('.k-carousel-arrow-next')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: arrow is hover', async () => {
		// @ts-ignore
		const instance = new KCarouselHover({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('.k-carousel-arrow-prev')).not.toBeTruthy();
		expect(host.querySelector('.k-carousel-arrow-next')).not.toBeTruthy();
		const carouselElm = host.querySelector('.k-carousel');
		carouselElm.dispatchEvent(new Event('mouseenter', { bubbles: true }));
		await tick();
		expect(host.querySelector('.k-carousel-arrow-prev')).toBeTruthy();
		expect(host.querySelector('.k-carousel-arrow-next')).toBeTruthy();
	});

	test('props: arrow is never', async () => {
		// @ts-ignore
		const instance = new KCarouselNever({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('.k-carousel-arrow-prev')).not.toBeTruthy();
		expect(host.querySelector('.k-carousel-arrow-next')).not.toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: loop', async () => {
		// @ts-ignore
		const instance = new KCarouselLoop({
			target: host
		});
		expect(instance).toBeTruthy();
		const carouselElm1 = host.querySelector('[test-id="loop_open"]');
		const carouselElm2 = host.querySelector('[test-id="loop_close"]');
		expect(carouselElm1.querySelector('[data-active="2"]')).toBeTruthy();
		expect(carouselElm2.querySelector('[data-active="2"]')).toBeTruthy();

		const nextBtn1 = carouselElm1.querySelector('.k-carousel-arrow-next');
		const nextBtn2 = carouselElm2.querySelector('.k-carousel-arrow-next');
		nextBtn1.click();
		await tick();
		await vi.advanceTimersByTimeAsync(300);
		expect(carouselElm1.querySelector('[data-active="0"]')).toBeTruthy();
		nextBtn2.click();
		await tick();
		await vi.advanceTimersByTimeAsync(300);
		expect(carouselElm2.querySelector('[data-active="2"]')).toBeTruthy();

		const prevBtn1 = carouselElm1.querySelector('.k-carousel-arrow-prev');
		prevBtn1.click();
		await tick();
		await vi.advanceTimersByTimeAsync(300);
		expect(carouselElm1.querySelector('[data-active="2"]')).toBeTruthy();
	});

	test('props: autoplay & interval & pauseOnHover', async () => {
		// @ts-ignore
		const instance = new KCarouselPlay({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('[data-active="1"]')).toBeTruthy();
		await tick();
		await vi.advanceTimersByTimeAsync(300);
		expect(host.querySelector('[data-active="2"]')).toBeTruthy();
		await tick();
		await vi.advanceTimersByTimeAsync(300);
		expect(host.querySelector('[data-active="0"]')).toBeTruthy();
		await tick();
		await vi.advanceTimersByTimeAsync(300);
		expect(host.querySelector('[data-active="1"]')).toBeTruthy();

		const carouselElm = host.querySelector('.k-carousel');
		carouselElm.dispatchEvent(new Event('mouseenter', { bubbles: true }));
		await tick();
		await vi.advanceTimersByTimeAsync(600);
		expect(host.querySelector('[data-active="1"]')).toBeTruthy();
	});

	test('api: goto', async () => {
		// @ts-ignore
		const instance = new KCarouselApiGoto({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('[data-active="1"]')).toBeTruthy();
		const btn = host.querySelector('button');
		btn.click();
		await tick();
		expect(host.querySelector('[data-active="2"]')).toBeTruthy();
	});

	test('api: prev', async () => {
		// @ts-ignore
		const instance = new KCarouselApiPrev({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('[data-active="1"]')).toBeTruthy();
		const btn = host.querySelector('button');
		btn.click();
		await tick();
		expect(host.querySelector('[data-active="0"]')).toBeTruthy();
	});

	test('api: next', async () => {
		// @ts-ignore
		const instance = new KCarouselApiNext({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('[data-active="1"]')).toBeTruthy();
		const btn = host.querySelector('button');
		btn.click();
		await tick();
		expect(host.querySelector('[data-active="2"]')).toBeTruthy();
	});
	test('event: change event caused by arrow and indicators', async () => {
		// @ts-ignore
		const instance = new KCarouselEvt({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.querySelector('[data-active="1"]')).toBeTruthy();
		const indicatorsElms = host.querySelectorAll('.k-indicators-item');
		indicatorsElms[2].click();
		await tick();
		expect(host.innerHTML).toMatchSnapshot('{"index":2,"oldIndex":1}');
		const nextBtn = host.querySelector('.k-carousel-arrow-next');
		nextBtn.click();
		await tick();
		expect(host.innerHTML).toMatchSnapshot('{"index":1,"oldIndex":2}');
		const prevBtn = host.querySelector('.k-carousel-arrow-prev');
		prevBtn.click();
		await tick();
		expect(host.innerHTML).toMatchSnapshot('{"index":2,"oldIndex":1}');
	});

	test('event: change event caused by auto play', async () => {
		// @ts-ignore
		const instance = new KCarouselEvtPlay({
			target: host
		});
		expect(instance).toBeTruthy();
		await tick();
		await vi.advanceTimersByTimeAsync(300);
		expect(host.innerHTML).toMatchSnapshot('{"index":2,"oldIndex":1}');
		await tick();
		await vi.advanceTimersByTimeAsync(300);
		expect(host.innerHTML).toMatchSnapshot('{"index":1,"oldIndex":2}');
	});

	test('event: change event caused by api', async () => {
		// @ts-ignore
		const instance = new KCarouselEvtApi({
			target: host
		});
		expect(instance).toBeTruthy();
		const btnGoto = host.querySelector('#btn_goto');
		btnGoto.click();
		await tick();
		expect(host.innerHTML).toMatchSnapshot('{"index":2,"oldIndex":1}');

		const btnNext = host.querySelector('#btn_next');
		btnNext.click();
		await tick();
		expect(host.innerHTML).toMatchSnapshot('{"index":0,"oldIndex":2}');

		const btnPrev = host.querySelector('#btn_prev');
		btnPrev.click();
		await tick();
		expect(host.innerHTML).toMatchSnapshot('{"index":2,"oldIndex":0}');
	});

	test('slot: custom arrow indicators ', async () => {
		// @ts-ignore
		const instance = new KCarouselCustom({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});
});
