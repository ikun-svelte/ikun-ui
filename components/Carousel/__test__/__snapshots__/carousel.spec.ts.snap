// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KCarousel > event: change event caused by api > {"index":0,"oldIndex":2} 1`] = `"<div aria-hidden="true" class="k-carousel"><div class="k-carousel-wrap" style="height: 0px;"><div class="k-carousel-item-wrap" data-active="0" data-carousel-container="" style="width: 300%; left: -300%; transition: left .5s ease 0s;"><div class="bg-green-400/15" style="height: 280px; transform: translateX(300%);"></div> <div class="bg-red-500/15" style="height: 280px;"></div> <div class="bg-blue-600/15" style="height: 280px;"></div></div></div> <div class="k-carousel-arrow"> </div> <div class="k-indicators"><div role="button" class="k-indicators-item k-indicators-item--active" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div></div></div> <div>{"index":2,"oldIndex":1}</div> <button id="btn_goto">goto</button> <button id="btn_prev">prev</button> <button id="btn_next">next</button>"`;

exports[`Test: KCarousel > event: change event caused by api > {"index":2,"oldIndex":0} 1`] = `"<div aria-hidden="true" class="k-carousel"><div class="k-carousel-wrap" style="height: 0px;"><div class="k-carousel-item-wrap" data-active="3" data-carousel-container="" style="width: 300%; left: 100%; transition: left .5s ease 0s;"><div class="bg-green-400/15" style="height: 280px; transform: translateX(300%);"></div> <div class="bg-red-500/15" style="height: 280px;"></div> <div class="bg-blue-600/15" style="height: 280px; transform: translateX(-300%);"></div></div></div> <div class="k-carousel-arrow"> </div> <div class="k-indicators"><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div></div></div> <div>{"index":2,"oldIndex":1}</div> <button id="btn_goto">goto</button> <button id="btn_prev">prev</button> <button id="btn_next">next</button>"`;

exports[`Test: KCarousel > event: change event caused by api > {"index":2,"oldIndex":1} 1`] = `"<div aria-hidden="true" class="k-carousel"><div class="k-carousel-wrap" style="height: 0px;"><div class="k-carousel-item-wrap" data-active="2" data-carousel-container="" style="width: 300%; left: -200%; transition: left .5s ease 0s;"><div class="bg-green-400/15" style="height: 280px;"></div> <div class="bg-red-500/15" style="height: 280px;"></div> <div class="bg-blue-600/15" style="height: 280px;"></div></div></div> <div class="k-carousel-arrow"> </div> <div class="k-indicators"><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item k-indicators-item--active" aria-hidden="true"></div></div></div> <div>{"index":2,"oldIndex":1}</div> <button id="btn_goto">goto</button> <button id="btn_prev">prev</button> <button id="btn_next">next</button>"`;

exports[`Test: KCarousel > event: change event caused by arrow and indicators > {"index":1,"oldIndex":2} 1`] = `"<div aria-hidden="true" class="k-carousel"><div class="k-carousel-wrap" style="height: 0px;"><div class="k-carousel-item-wrap" data-active="0" data-carousel-container="" style="width: 300%; left: -300%; transition: left .5s ease 0s;"><div class="bg-green-400/15" style="height: 280px; transform: translateX(300%);"></div> <div class="bg-red-500/15" style="height: 280px;"></div> <div class="bg-blue-600/15" style="height: 280px;"></div></div></div> <div class="k-carousel-arrow"><div role="button" class="k-carousel-arrow-icon k-carousel-arrow-icon--dark k-carousel-arrow-prev" aria-hidden="true"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-left" role="" aria-hidden="true" style="width: 16px; height: 16px;"></span></div> <div role="button" class="k-carousel-arrow-icon k-carousel-arrow-icon--dark k-carousel-arrow-next" aria-hidden="true"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-right" role="" aria-hidden="true" style="width: 16px; height: 16px;"></span></div></div> <div class="k-indicators"><div role="button" class="k-indicators-item k-indicators-item--active" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div></div></div> <div>{"index":2,"oldIndex":1}</div>"`;

exports[`Test: KCarousel > event: change event caused by arrow and indicators > {"index":2,"oldIndex":1} 1`] = `"<div aria-hidden="true" class="k-carousel"><div class="k-carousel-wrap" style="height: 0px;"><div class="k-carousel-item-wrap" data-active="2" data-carousel-container="" style="width: 300%; left: -200%; transition: left .5s ease 0s;"><div class="bg-green-400/15" style="height: 280px;"></div> <div class="bg-red-500/15" style="height: 280px;"></div> <div class="bg-blue-600/15" style="height: 280px;"></div></div></div> <div class="k-carousel-arrow"><div role="button" class="k-carousel-arrow-icon k-carousel-arrow-icon--dark k-carousel-arrow-prev" aria-hidden="true"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-left" role="" aria-hidden="true" style="width: 16px; height: 16px;"></span></div> <div role="button" class="k-carousel-arrow-icon k-carousel-arrow-icon--dark k-carousel-arrow-next" aria-hidden="true"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-right" role="" aria-hidden="true" style="width: 16px; height: 16px;"></span></div></div> <div class="k-indicators"><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item k-indicators-item--active" aria-hidden="true"></div></div></div> <div>{"index":2,"oldIndex":1}</div>"`;

exports[`Test: KCarousel > event: change event caused by arrow and indicators > {"index":2,"oldIndex":1} 2`] = `"<div aria-hidden="true" class="k-carousel"><div class="k-carousel-wrap" style="height: 0px;"><div class="k-carousel-item-wrap" data-active="3" data-carousel-container="" style="width: 300%; left: 100%; transition: left .5s ease 0s;"><div class="bg-green-400/15" style="height: 280px; transform: translateX(300%);"></div> <div class="bg-red-500/15" style="height: 280px;"></div> <div class="bg-blue-600/15" style="height: 280px; transform: translateX(-300%);"></div></div></div> <div class="k-carousel-arrow"><div role="button" class="k-carousel-arrow-icon k-carousel-arrow-icon--dark k-carousel-arrow-prev" aria-hidden="true"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-left" role="" aria-hidden="true" style="width: 16px; height: 16px;"></span></div> <div role="button" class="k-carousel-arrow-icon k-carousel-arrow-icon--dark k-carousel-arrow-next" aria-hidden="true"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-right" role="" aria-hidden="true" style="width: 16px; height: 16px;"></span></div></div> <div class="k-indicators"><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div></div></div> <div>{"index":2,"oldIndex":1}</div>"`;

exports[`Test: KCarousel > event: change event caused by auto play > {"index":1,"oldIndex":2} 1`] = `"<div aria-hidden="true" class="k-carousel"><div class="k-carousel-wrap" style="height: 0px;"><div class="k-carousel-item-wrap" data-active="0" data-carousel-container="" style="width: 300%; left: -300%; transition: left .5s ease 0s;"><div class="bg-green-400/15" style="height: 280px; transform: translateX(300%);"></div> <div class="bg-red-500/15" style="height: 280px;"></div> <div class="bg-blue-600/15" style="height: 280px;"></div></div></div> <div class="k-carousel-arrow"> </div> <div class="k-indicators"><div role="button" class="k-indicators-item k-indicators-item--active" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div></div></div> <div>{"index":2,"oldIndex":1}</div>"`;

exports[`Test: KCarousel > event: change event caused by auto play > {"index":2,"oldIndex":1} 1`] = `"<div aria-hidden="true" class="k-carousel"><div class="k-carousel-wrap" style="height: 0px;"><div class="k-carousel-item-wrap" data-active="2" data-carousel-container="" style="width: 300%; left: -200%; transition: left .5s ease 0s;"><div class="bg-green-400/15" style="height: 280px;"></div> <div class="bg-red-500/15" style="height: 280px;"></div> <div class="bg-blue-600/15" style="height: 280px;"></div></div></div> <div class="k-carousel-arrow"> </div> <div class="k-indicators"><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item k-indicators-item--active" aria-hidden="true"></div></div></div> <div>{"index":2,"oldIndex":1}</div>"`;

exports[`Test: KCarousel > props: arrow is always 1`] = `"<div aria-hidden="true" class="k-carousel"><div class="k-carousel-wrap" style="height: 0px;"><div class="k-carousel-item-wrap" data-active="1" data-carousel-container="" style="width: 300%; left: -100%; transition: left .5s ease 0s;"><div class="bg-green-400/15" style="height: 280px;"></div> <div class="bg-red-500/15" style="height: 280px;"></div> <div class="bg-blue-600/15" style="height: 280px;"></div></div></div> <div class="k-carousel-arrow"><div role="button" class="k-carousel-arrow-icon k-carousel-arrow-icon--dark k-carousel-arrow-prev" aria-hidden="true"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-left" role="" aria-hidden="true" style="width: 16px; height: 16px;"></span></div> <div role="button" class="k-carousel-arrow-icon k-carousel-arrow-icon--dark k-carousel-arrow-next" aria-hidden="true"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-right" role="" aria-hidden="true" style="width: 16px; height: 16px;"></span></div></div> <div class="k-indicators"><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item k-indicators-item--active" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div></div></div>"`;

exports[`Test: KCarousel > props: arrow is never 1`] = `"<div aria-hidden="true" class="k-carousel"><div class="k-carousel-wrap" style="height: 0px;"><div class="k-carousel-item-wrap" data-active="1" data-carousel-container="" style="width: 300%; left: -100%; transition: left .5s ease 0s;"><div class="bg-green-400/15" style="height: 280px;"></div> <div class="bg-red-500/15" style="height: 280px;"></div> <div class="bg-blue-600/15" style="height: 280px;"></div></div></div> <div class="k-carousel-arrow"> </div> <div class="k-indicators"><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item k-indicators-item--active" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div></div></div>"`;

exports[`Test: KCarousel > props: cls 1`] = `"<div aria-hidden="true" class="k-carousel k-carousel--test"><div class="k-carousel-wrap"><div class="k-carousel-item-wrap" data-active="0" data-carousel-container="" style="width: 300%; left: -0%; transition: left .5s ease 0s;"></div></div> <div class="k-carousel-arrow"> </div> <div class="k-indicators"><div role="button" class="k-indicators-item k-indicators-item--active" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div></div></div>"`;

exports[`Test: KCarousel > props: initialIndex 1`] = `"<div aria-hidden="true" class="k-carousel"><div class="k-carousel-wrap" style="height: 0px;"><div class="k-carousel-item-wrap" data-active="1" data-carousel-container="" style="width: 300%; left: -100%; transition: left .5s ease 0s;"><div class="bg-green-400/15" style="height: 280px;"></div> <div class="bg-red-500/15" style="height: 280px;"></div> <div class="bg-blue-600/15" style="height: 280px;"></div></div></div> <div class="k-carousel-arrow"> </div> <div class="k-indicators"><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item k-indicators-item--active" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div></div></div>"`;

exports[`Test: KCarousel > slot: custom arrow indicators  1`] = `"<div aria-hidden="true" class="k-carousel" style="height: 300px;"><div class="k-carousel-wrap" style="height: 0px;"><div class="k-carousel-item-wrap" data-active="0" data-carousel-container="" style="width: 600%; left: -0%; transition: left .5s ease 0s;"><div class="bg-orange-100 fcc" style="height: 280px;"><h1>0</h1></div> <div class="bg-orange-200 fcc" style="height: 280px;"><h1>1</h1></div> <div class="bg-orange-100 fcc" style="height: 280px;"><h1>2</h1></div> <div class="bg-orange-200 fcc" style="height: 280px;"><h1>3</h1></div> <div class="bg-orange-100 fcc" style="height: 280px;"><h1>4</h1></div> <div class="bg-orange-200 fcc" style="height: 280px;"><h1>5</h1></div></div></div> <div slot="arrow" class="flex"><div class="k-carousel-arrow"><div aria-hidden="true" class="ml-10px cursor-pointer" slot="prev">🏀</div> <div aria-hidden="true" class="mr-10px cursor-pointer" slot="next">🏀</div></div></div> <div slot="indicators"><div class="k-carousel-indicators demo-indicators"><div slot="item" class="flex"><div class="demo-carousel-item demo-carousel-item--active svelte-1c5n2gd" role="button" aria-hidden="true"></div><div class="demo-carousel-item svelte-1c5n2gd" role="button" aria-hidden="true"></div><div class="demo-carousel-item svelte-1c5n2gd" role="button" aria-hidden="true"></div><div class="demo-carousel-item svelte-1c5n2gd" role="button" aria-hidden="true"></div><div class="demo-carousel-item svelte-1c5n2gd" role="button" aria-hidden="true"></div><div class="demo-carousel-item svelte-1c5n2gd" role="button" aria-hidden="true"></div></div></div></div></div>"`;
